### ISSUE TITLE
headscale.service location

### ISSUE THREAD

**Issue by zoenglinghou** [juanfont/headscale#1359]
**Bug description**
`headscale.service` should not be installed to `/etc/systemd/system`

**To Reproduce**

```bash
dpkg -L headscale.deb
```

**Context info**
It is my understanding that systemd `unit` files from a package should be installed to `/usr/lib/systemd/system`, and would be symlinked to `/etc/systemd/system` upon `systemctl enable`. Sometimes the `postinst` and `prerm` would handle the service enabling and stopping too.

**Comment by jonathanspw** [comment]
You are correct on this.

**Comment by loprima-l** [comment]
I guess it depends on the linux version and OS, I'm not so familiar with Linux, did you know how is it on other OSes?

**Comment by zoenglinghou** [comment]
> I guess it depends on the linux version and OS, I'm not so familiar with Linux, did you know how is it on other OSes?

I know this is the packaging convention on Debian-, RedHat-, and Arch-Families. macOS would use a different init system

**Comment by jonathanspw** [comment]
It's standard across systemd-based systems.  `/usr/lib/systemd` is for units from packages.  `/etc/systemd/system` is for custom stuff/non-packaged stuff.

> In addition to /etc/systemd/system, the drop-in ".d/" directories for system services can be placed in /usr/lib/systemd/system or /run/systemd/system directories. Drop-in files in /etc/ take precedence over those in /run/ which in turn take precedence over those in /usr/lib/. Drop-in files under any of these directories take precedence over unit files wherever located. Multiple drop-in files with different names are applied in lexicographic order, regardless of which of the directories they reside in.

src: https://www.freedesktop.org/software/systemd/man/systemd.unit.html

tldr: it's all about precedence.  Use /usr/lib/systemd to allow users to override things if they desire in a clean way.

**Comment by loprima-l** [comment]
@jonathanspw I think you're already working on the RPM package build, as I'm not familiar with Linux, can you make a pull request for this or at least make the changes on the .rpm ?

**Comment by jonathanspw** [comment]
It's already correct in the RPMs.  PRd to fix the `.deb`s.

**Comment by loprima-l** [comment]
Oh, I supposed that the error was everywhere, nice job ^^

**Comment by loprima-l** [comment]
@zoenglinghou You can close your Issue, @jonathanspw  Got ya !

**Comment by jonathanspw** [comment]
When the PR is merged it will close it automatically :)

**Comment by zoenglinghou** [comment]
Thanks @jonathanspw
